persistence/cosmosdb/index_outboxstorageformat_CosmosDB_[3.2,).partial.md

```diff
@@ -0,0 +1,9 @@
+A default [synthetic partition key](https://learn.microsoft.com/en-us/azure/cosmos-db/nosql/synthetic-partition-keys) will be used for all incoming messages, in the format `{endpointName}-{messageId}`, if not explicitly [overwritten](/persistence/cosmosdb/transactions.md#specifying-the-partitionkey-to-use-for-the-transaction) at runtime.
+
+> [!NOTE]
+> The [default partition key must be overwritten](/persistence/cosmosdb/transactions.md#specifying-the-partitionkey-to-use-for-the-transaction) whenever the message handler creates or updates business documents in CosmosDB. To guarantee atomicity, explicitly set the Outbox partition key to match the partition key of your business document (e.g., an `OrderId` or `CustomerId`). This is a requirement for including both the business document and the Outbox record in the same [Cosmos DB transactional batch](https://learn.microsoft.com/en-us/azure/cosmos-db/partitioning-overview). Conversely, for simplicity, you can use the default partition key when a handler's logic does not involve persisting business data.
+
+To support backward compatibility of control messages during migration, the persistence includes a fallback mechanism. When enabled (default), and if a record is not found using the synthetic key format, the system falls back to the legacy `{messageId}` format. The fallback can be disabled using `DisableReadFallback()` once all legacy records have expired, which is recommended since the fallback mechanism involves an additional read operation on the Outbox container.
+
+> [!WARNING]
+> When the default synthetic partition key is not used, but rather overwritten at runtime, Outbox records are not automatically separated by endpoint name. As a result, multiple logical endpoints cannot share the same database and container since [message identities are not unique across endpoints from a processing perspective](/nservicebus/outbox/#message-identity). To avoid conflicts, either separate different endpoints into different containers, use the default synthetic partition key, or [override the default synthetic partition key](transactions.md) in a way that ensures message identities are unique to each processing endpoint.
```

Comment:
Wouldn't this be dependant on the key used? Ie if they use a key that includes the endpoint name then it's not the case. So perhaps something like: 

```suggestion
>  Since [message identities are not unique across endpoints from a processing perspective](/nservicebus/outbox/#message-identity), when overwriting the default synthetic key, either separate different endpoints into different containers or [override the default synthetic partition key](transactions.md) in a way that ensures message identities are unique to each processing endpoint.
```